saltboot-formula/_states/saltboot.py

```diff
@@ -1770,12 +1770,14 @@ def bootloader_updated(name, partitioning, images, boot_images, terminal_kernel_
             #
             # failures do matter only if kernel version differs and kexec fails, so do not set ret['result'] here
             try:
-                res = __salt__['cmd.run_all'](_download_url_cmd(boot_image.get('kernel', {}).get('url', ''), os.path.join(prefix, 'boot/Image')))
+                cmd = _download_url_cmd(boot_image.get('kernel', {}).get('url', ''), os.path.join(prefix, 'boot/Image'))
+                res = __salt__['cmd.run_all'](cmd)
                 if res['retcode'] > 0:
-                    ret['comment'] += res['stdout'] + res['stderr']
-                res = __salt__['cmd.run_all'](_download_url_cmd(boot_image.get('initrd', {}).get('url', ''), os.path.join(prefix, 'boot/initrd')))
+                    ret['comment'] += cmd + " : " + res['stdout'] + res['stderr'] + "\n"
```

Comment:
Some comment may be good here.
Remember how the failure looked like for @lkotek?
```
salt_device=/dev/mmcblk1p4
boot_image=POS_Image_JeOS7-7.0.0
root=/dev/mmcblk1p4
terminal_kernel_parameters=Nonetftp: server error: (1) File not foundtftp: server error: (1) File not found
```

So this will look mangled as well.
```
terminal_kernel_parameters=Nonebusybox tftp .... tftp: server error: (1) File not foundtftp: server error: (1) File not found
```

At least start with new line and add some explanation - 'Kernel download failed:'